src/coreclr/jit/emitxarch.cpp

```diff
@@ -2012,16 +2036,25 @@ bool emitter::TakesApxExtendedEvexPrefix(const instrDesc* id) const
         return false;
     }
 
-    if (id->idIsEvexNdContextSet())
+    if (id->idIsEvexNdContextSet() && HasApxNdd(ins))
     {
+        // The instruction uses APX-ND hint, and it requires EVEX.
         return true;
     }
 
-    if (id->idIsEvexNfContextSet())
+    if (id->idIsEvexNfContextSet() && HasApxNf(ins))
     {
+        // The instruction uses APX-NF hint, and it requires EVEX.
         return true;
     }
 
+    if (IsSETZUccInstruction(ins))
+    {
+        // These are promoted forms of SETcc instruction with EVEX.ZU.
+        // TODO-XArch-APX: maybe consider return true as we may only use those instructions with ZU set.
```

Comment:
I would expect us to either:
A. Only have `INS_setcc` where that is used for legacy or VEX and `zu` is recognized by the `EVEX.zu` context bit
-or-
B. We have both `INS_setcc` and `INS_setzucc`, with the former being exclusively for legacy/rex2 encoding and the latter being exclusively for EVEX encoding

Given it looks like `setzucc` has a different set of opcodes, I would expect we need to go with option B and so context bit is potentially unnecessary.